app/forms/citizens/other_assets_form.rb

```diff
@@ -0,0 +1,104 @@
+module Citizens
+  class OtherAssetsForm
+    include BaseForm
+
+    form_for OtherAssetsDeclaration
+
+    SINGLE_VALUE_ATTRIBUTES = %i[
+      timeshare_value
+      land_value
+      jewellery_value
+      vehicle_value
+      classic_car_value
+      money_assets_value
+      money_owed_value
+      trust_value
+    ].freeze
+
+    SECOND_HOME_ATTRIBUTES = %i[
+      second_home_value
+      second_home_mortgage
+      second_home_percentage
+    ].freeze
+
+    ALL_ATTRIBUTES = (SECOND_HOME_ATTRIBUTES + SINGLE_VALUE_ATTRIBUTES).freeze
+
+    CHECK_BOXES_ATTRIBUTES = (SINGLE_VALUE_ATTRIBUTES.map { |attribute| "check_box_#{attribute}".to_sym } + [:check_box_second_home]).freeze
+
+    SINGLE_VALUE_ATTRIBUTES.each do |attribute|
+      check_box_attribute = "check_box_#{attribute}".to_sym
+      validates attribute, numericality: { greater_than_or_equal_to: 0 }, allow_blank: true
+      validates attribute, presence: true, if: proc { |form| form.send(check_box_attribute).present? }
+    end
+
+    validates :second_home_value, :second_home_mortgage, :second_home_percentage, numericality: { greater_than_or_equal_to: 0 }, allow_blank: true
+
+    attr_accessor(*ALL_ATTRIBUTES)
+    attr_accessor(*CHECK_BOXES_ATTRIBUTES)
+    attr_accessor :check_box_second_home
+
+    before_validation :normalize_amounts, :empty_unchecked_values
+
+    validate :all_second_home_values_present_if_any_are_present
+
+    def second_home_checkbox_status
+      any_second_home_value_present? ? 'checked' : nil
+    end
+
+    def any_second_home_value_present?
+      present_and_not_zero?(@second_home_value) ||
+        present_and_not_zero?(@second_home_mortgage) ||
+        present_and_not_zero?(@second_home_percentage)
+    end
+
+    def exclude_from_model
+      CHECK_BOXES_ATTRIBUTES
+    end
+
+    private
+
+    def empty_unchecked_values
+      SINGLE_VALUE_ATTRIBUTES.each do |attribute|
+        check_box_attribute = "check_box_#{attribute}".to_sym
+        if send(check_box_attribute).blank?
+          attributes[attribute] = nil
+          send("#{attribute}=", nil)
+        end
+      end
+
+      nullify_second_home_attrs if @check_box_second_home.blank?
+    end
+
+    def nullify_second_home_attrs
+      attributes[:second_home_value] = nil
+      attributes[:second_home_mortgage] = nil
+      attributes[:second_home_percentage] = nil
+      @second_home_value = nil
+      @second_home_mortgage = nil
+      @second_home_percentage = nil
+    end
+
+    def normalize_amounts
+      ALL_ATTRIBUTES
+        .map { |attribute| send(attribute) }
+        .compact
+        .each { |value| value.delete!(',') }
+    end
+
+    def all_second_home_values_present_if_any_are_present
+      return unless any_second_home_value_present?
+
+      %i[second_home_value second_home_mortgage second_home_percentage].each do |attr|
+        errors.add(attr, :blank) if __send__(attr).blank?
```

Comment:
I think `send` and `__send__` are the same thing in this object. So I think we should be consistent and use one in the object code.